### ISSUE TITLE
Src file mapping not working properly with `trybuild` and proc-macro workspace member 

### ISSUE THREAD

**Issue by teoxoy** [taiki-e/cargo-llvm-cov#140]
When running integration tests (via [`trybuild`](https://github.com/dtolnay/trybuild)) for a proc-macro crate that is a workspace member, the following warning is output:

```
error: dep/dep/src/lib.rs: No such file or directory
warning: The file 'dep/dep/src/lib.rs' isn't covered.
```

cmd run:

`cargo llvm-cov --workspace --html`

The report ends up showing 2 entries for the same src file as seen below

![image](https://user-images.githubusercontent.com/28601907/155600356-d125ebb3-4c40-4e70-b614-c07d1cd3846c.png)

You can find a small reproducible sample here: [test.tar.gz](https://github.com/taiki-e/cargo-llvm-cov/files/8135449/test.tar.gz)

This seems to be happening even if `trybuild::TestCases::new()` is the only thing in the test fn (no calls to `pass` or `compile_fail`).
It also seems to work fine if the crate is a normal lib instead of a proc-macro lib.

I wasn't sure if this is an issue with `cargo-llvm-cov` or `trybuild` but having tried [`tarpaulin`](https://github.com/xd009642/tarpaulin) with success (same setup), I thought it might be an issue with `cargo-llvm-cov`.

**Comment by taiki-e** [comment]
Thanks for the report!

This appears to be a bug in rustc's --remap-path-prefix. If I disable it, it works correctly (https://github.com/taiki-e/cargo-llvm-cov/pull/141).

<img width="869" alt="m" src="https://user-images.githubusercontent.com/43724913/155658438-9e53fd93-a591-4b97-b6b7-9088b32b690f.png">

**Comment by teoxoy** [comment]
I tested the PR and it seems to solve the issue.
Thanks for the quick response and fix!

Do you have any concerns with the change in the PR?
Also, should we report the bug upstream?

**Comment by taiki-e** [comment]
The problem I am currently aware of is that our test suite does not work well because the reports show absolute paths.

https://github.com/taiki-e/cargo-llvm-cov/blob/20d5da7cab721cdd3d140beb86d48e8076f291a5/tests/fixtures/coverage-reports/no_test/link_dead_code.txt#L1

Another concern is that this may be a problem when merging coverage between different platforms. (I have not looked into this yet.)

So, for now, I'm starting to think it is reasonable to provide an option to disable --remap-path-prefix.

**Comment by taiki-e** [comment]
According to https://github.com/ruifengx/rowantlr/pull/3, this seems to be a regression introduced around 2021-11-30.

**Comment by teoxoy** [comment]
Maybe something to do with https://github.com/rust-lang/rust/pull/83846 (merged on 2021-11-11)?

**Comment by taiki-e** [comment]
Published 0.3.0 which includes the fix for this.
